### PR TITLE
Refactor nnictl to support list multiple experiment

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: tools/nnicmd/config_utils.py [hunk 1]
```diff
@@ -71,3 +65,44 @@ def read_file(self):
             except ValueError:
                 return {}
         return {}
+
+class Experiment:
```
Comment (QuanluZhang) [inline tools/nnicmd/config_utils.py]: if this class is for maintaining experiment list, why not rename `Experiment` to `Experiments`?

File: tools/nnicmd/config_utils.py [hunk 2]
```diff
@@ -71,3 +65,44 @@ def read_file(self):
...
+    '''Maintain experiment list'''
+    def __init__(self):
+        os.makedirs(HOME_DIR, exist_ok=True)
+        self.experiment_file = os.path.join(HOME_DIR, '.experiment')
+        self.experiment = self.read_file()
```
Comment (QuanluZhang) [inline tools/nnicmd/config_utils.py]: similarly, `self.experiment` -> `self.experiments`

**Comment by SparkSnail** [comment]
https://github.com/Microsoft/nni/issues/213
https://github.com/Microsoft/nni/issues/178
https://github.com/Microsoft/nni/issues/177
https://github.com/Microsoft/nni/issues/176

File: tools/nnicmd/config_utils.py [hunk 3]
```diff
@@ -71,3 +65,44 @@ def read_file(self):
             except ValueError:
                 return {}
         return {}
+
+class Experiments:
+    '''Maintain experiment list'''
+    def __init__(self):
+        os.makedirs(HOME_DIR, exist_ok=True)
+        self.experiment_file = os.path.join(HOME_DIR, '.experiment')
```
Comment (yds05) [inline tools/nnicmd/config_utils.py]: how about move .experiment to HOME_DIR/nni?

File: tools/nnicmd/launcher.py [hunk 4]
```diff
@@ -53,8 +58,12 @@ def start_rest_server(port, platform, mode, experiment_id=None):
     stderr_full_path = os.path.join(HOME_DIR, str(port), 'stderr')
     stdout_file = open(stdout_full_path, 'a+')
     stderr_file = open(stderr_full_path, 'a+')
+    time_now = time.strftime('%Y-%m-%d %H:%M:%S',time.localtime(time.time()))
+    log_header = LOG_HEADER % str(time_now)
```
Comment (yds05) [inline tools/nnicmd/launcher.py]: Could you please add comment to explain what's the use of this log header format?

File: tools/nnicmd/nnictl_utils.py [hunk 5]
```diff
@@ -24,12 +24,34 @@
...
-from .config_utils import Config
+from .config_utils import Config, Experiments
 from .url_utils import trial_jobs_url, experiment_url, trial_job_id_url
-from .constants import HOME_DIR
+from .constants import HOME_DIR, EXPERIMENT_ID_INFO
 import time
 from .common_utils import print_normal, print_error, detect_process
 
+def get_experiment_port(args):
+    experiment_config = Experiments()
```
Comment (yds05) [inline tools/nnicmd/nnictl_utils.py]: Add function document description here

File: tools/nnicmd/nnictl_utils.py [hunk 6]
```diff
@@ -44,41 +66,95 @@ def convert_time_stamp_to_date(content):
...
+def parse_ids(args):
+    '''Parse the arguments for nnictl stop'''
+    experiment_config = Experiments()
+    experiment_dict = experiment_config.get_all_experiments()
+    if not experiment_dict:
+        print_normal('Experiment is not running...')
+        return None
+    experiment_id_list = list(experiment_dict.keys())
+    result_list = []
+    if not args.id:
```
Comment (yds05) [inline tools/nnicmd/nnictl_utils.py]: Add comment to explain the rule which you used to parse id?

File: tools/nnicmd/config_utils.py (hunk 3 above)
Comment (yds05) [reply tools/nnicmd/config_utils.py]: I see. So suggest to rename HOME_DIR to NNICTL_HOME_DIR to dispense with any confusion.

**Pull request merged**